Src/LinearSolvers/MLMG/AMReX_MLEBABecLap.cpp

```diff
@@ -185,14 +185,22 @@ MLEBABecLap::setBCoeffs (int amrlev, Vector<Real> const& beta)
 void
 MLEBABecLap::setEBDirichlet (int amrlev, const MultiFab& phi, const MultiFab& beta)
 {
+    bool  phi_on_centroid = (m_phi_loc  == Location::CellCentroid);
+
+    if (phi_on_centroid) {
+      // If we call setEBDirichlet instead of setEBHomogDirichlet then we assume posisbly inhomogeneous
```

Comment:
Don't think this is needed.  `m_is_eb_inhomog` is a `mutable` that is set in `applyBC`.     Then in `Fapply` and `getEBFluxes`, the value of `m_is_eb_inhomog` is used.  It is actaully false (homogeneous) for inhomogeneous Drichlet EB when the multigrid solver is in correct form.

Reply:
thanks for pointing this out. currently the unit test calls [MLMG::apply()](https://github.com/drangara/amrex/blob/eb_viscous_explicit/Tests/LinearSolvers/LeastSquares/MyTest.cpp#L322)
which seems to call the `MLCellLinOp::solutionResidual()` by setting `inhomogeneous=true`. When `MLEBABecLap::applyBC()` is called, the inhomog flag is always true. I think removing the statements which set `m_is_eb_inhomog=false` inside `setEBHomogDirichlet` is therefore causing these tests to fail. Not sure if perhaps the test setup itself needs to change to use the "correction" form?

Reply:
Maybe there is a bug in the development branch or your PR branch.  But I think the logic in the development branch is not wrong.  Function `setEBHomogDirichlet` sets `phi` on EB to zero.  So setting `m_is_eb_inhomog=true` is not wrong for homogeneous Dirichlet because inhomogeneous boundary with a Dirichlet value of zero is homogeneous Dirichlet.

Reply:
ah i see. let me check if there is a bug on my branch that is causing the failure when the flag is true. sounds like it should be able to handle the case since phi on EB should be set to 0. thank you.

Reply:
moved the handling of `inhomog=true` into the slopes methods and removed setting it in `setEBHomogDirichlet` functions.